### ISSUE TITLE
Dropdown should have both label for and aria-labelledby

### ISSUE THREAD

**Issue by silviuaavram** [microsoft/fluent-ui-react#1583]
`<ul>` should keep `aria-labelledby`
`<label>` should have `for` so clicking it sends focus to the widget.

At the moment both `<ul>` and `<input>/<button>` are `aria-labelledby` which makes label clicking not work.
